libraries/AP_Math/AP_Math.h

```diff
@@ -25,273 +17,115 @@
 #include "quaternion.h"
 #include "polygon.h"
 #include "edc.h"
-#include "float.h"
-#include <AP_Param/AP_Param.h>
+#include "location.h"
 
-#ifndef M_PI_F
- #define M_PI_F 3.141592653589793f
-#endif
-#ifndef M_2PI_F
- #define M_2PI_F (2*M_PI_F)
-#endif
-#ifndef PI
- # define PI M_PI_F
-#endif
-#ifndef M_PI_2
- # define M_PI_2 1.570796326794897f
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+    #include "AP_Math_Linux.h"
+#else
+    #include "AP_Math_PX4.h"
```

Comment:
I think we won't be able to do this to depend on board. I tried to convince other people about it on last dev call, but @tridge said he wanted to have the same behavior on sitl and PX4. Otherwise a bug in sitl could not reproduce in PX4 and vice-versa.

So what I think could be made is: AP_Math_Float.h and AP_Math_Double.h.  And allow sitl and linux boards to work with any of them (an option to waf configure would be fine IMO and default that to float). PX4 would work only with float, so that should be the default. He said however there are very little places benefiting from double-precision math... so he may want to chime in here. I think that being able to switch between one implementation and another would solve his concerns.

@tridge ? These are some examples where @dgrat's implementation increased the precision: https://github.com/diydrones/ardupilot/pull/3413#issuecomment-169350830 and https://github.com/diydrones/ardupilot/pull/3413#issuecomment-169353020